HIP/hip-1340.md

```diff
@@ -0,0 +1,376 @@
+
+In addition to the gas cost for each authorization list item (both valid and invalid) defined in EIP-7702 - 12500 or 25000 gas depending on whether the EOA alredy exists - Hedera will charge hollow account creation cost of TODO for any newly created accounts. The maximum number of created hollow acounts is limited by transaction gas limit as well as the ops duration throttle.
+
+Following EIP-7702, all invalid authorization list items will be silently ignored. If, however, an otherwise valid authorization list item cannot be processed because of gas shortage (e.g. if it results in hollow account creation) a child CryptoCreate (or CryptoUpdate) dispatch should still be created with a failed status to indicate the error to the user.
+
+The authorizations of an EVM type 4 transaction are processed before and independently of the subsequent EVM code execution, which might be a part of the same transaction. Any valid authorizations created while processing the list should be persisted, even if the subsequent EVM code execution fails/reverts.
+
+#### Updating Code Delegation - HSCS/HTS interoperability
+
+The diagram below illustratest the interoperability between HSCS and HTS for two cases: when the Code Delegation update originates from either EthereumTransaction or CryptoUpdate.
+
+![HSCS/HTS interoperability](../assets/hip-1340/hscs_hts_interoperability.png)
+
+#### Updating EOA's Code Delegation - Summary of Record Stream Changes
+
+- CryptoCreate and CryptoUpdate transaction bodies will be extended with a new field
+- state_changes will contain a `STATE_ID_BYTECODE` state change for each updated Delegation Indicator
+
+### Executing The Code Delegation Logic When an EVM Call Is Processed
+
+The second aspect of EOA Code Delegation is an update to the EVM call processing logic.
+
+When a call is made to an account - either a top-level EVM transaction call (i.e. when `transaction.to == EOA address`) or an inner call (e.g. `address(eoa).call(..)` call in a Smart Contract) - HSCS will:
+
+1. Fetch the bytecode of the target account of the call
+2. Recognize the Delegation Indicator by inspecting the prefix of the code for `0xef0100`
+3. Resolve the contract account that is the target of the delegation and run its EVM code in the context of the original EOA, subject to the limitations described below
+
+One notable difference is that EVM code execution might result in EVM storage updates on the original EOA account as well as emit EVM events.
+As a result, all queries related to EVM storage and logs previously available for contract accounts should now also function for EOAs, which include:
+- eth_getStorageAt RPC method
+- eth_getLogs RPC method
+- all related APIs on the Mirror Node
```

Comment:
I suggest to more specfic and mention that /api/v1/contracts/*.*, should be able to answer to any EOA information request, as from EIP-7702 onwards, EOAs are treated as Smart contracts